File: CASE_Simple_Example_V4/Hardened/HAMR/src/main/component/HAMR/SW/ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.scala
// #Sireum

package HAMR.SW

import org.sireum._
import HAMR._

// This file will not be overwritten so is safe to edit
object ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor {

  def initialise(api: ResponseMonitor_Impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.put_Alert()
  }

  def handle_FlightPlan(api: ResponseMonitor_Impl_Operational_Api, value : SW.Mission): Unit = {
    api.logInfo("example handle_FlightPlan implementation")
    api.logInfo(s"received ${value}")
    // example api usage

    val apiUsage_FlightPlan: Option[SW.Mission] = api.get_FlightPlan()
    api.logInfo(s"Received on FlightPlan: ${apiUsage_FlightPlan}")
    val apiUsage_MissionCommand: Option[SW.RF_Msg_Impl] = api.get_MissionCommand()
    api.logInfo(s"Received on MissionCommand: ${apiUsage_MissionCommand}")
  }

  def handle_MissionCommand(api: ResponseMonitor_Impl_Operational_Api, value : SW.RF_Msg_Impl): Unit = {
    api.logInfo("example handle_MissionCommand implementation")
    api.logInfo(s"received ${value}")
  }

  def activate(api: ResponseMonitor_Impl_Operational_Api): Unit = { }

  def deactivate(api: ResponseMonitor_Impl_Operational_Api): Unit = { }

  def finalise(api: ResponseMonitor_Impl_Operational_Api): Unit = { }

  def recover(api: ResponseMonitor_Impl_Operational_Api): Unit = { }
}

File: CASE_Simple_Example_V4/Hardened/HAMR/hamr/ext-c/ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor/ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.c
#include <ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_api.h>
#include <ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.c", "", "HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_initialise_", 0);

  // examples of api setter and logging usage

  api_put_Alert__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF_LAST);

  api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF string("Example logInfo"));

  api_logDebug__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF string("Example logDebug"));

  api_logError__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF string("Example logError"));
}

Unit HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.c", "", "HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_finalise_", 0);
}

Unit HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_handle_FlightPlan_(
  STACK_FRAME
  HAMR_SW_Mission value) {
  DeclNewStackFrame(caller, "ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.c", "", "HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_handle_FlightPlan_", 0);

  DeclNewString(FlightPlanString);
  String__append(SF (String) &FlightPlanString, string("HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_handle_FlightPlan called"));
  api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor (SF (String) &FlightPlanString);

  DeclNewString(_str);
  String__append(SF (String) &_str, string("Received on FlightPlan: "));
  HAMR_SW_Mission_string_(SF (String) &_str, value);
  api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF (String) &_str);

  // examples of api getter usage

  DeclNewHAMR_SW_Mission(t0);
  if(api_get_FlightPlan__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF &t0)) {
    DeclNewString(FlightPlan_str);
    String__append(SF (String) &FlightPlan_str, string("Received on FlightPlan: "));
    HAMR_SW_Mission_string_(SF (String) &FlightPlan_str, &t0);
    api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF (String) &FlightPlan_str);
  }

  DeclNewHAMR_SW_RF_Msg_Impl(t1);
  if(api_get_MissionCommand__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF &t1)) {
    DeclNewString(MissionCommand_str);
    String__append(SF (String) &MissionCommand_str, string("Received on MissionCommand: "));
    HAMR_SW_RF_Msg_Impl_string_(SF (String) &MissionCommand_str, &t1);
    api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF (String) &MissionCommand_str);
  }
}

Unit HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_handle_MissionCommand_(
  STACK_FRAME
  HAMR_SW_RF_Msg_Impl value) {
  DeclNewStackFrame(caller, "ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor.c", "", "HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_handle_MissionCommand_", 0);

  DeclNewString(MissionCommandString);
  String__append(SF (String) &MissionCommandString, string("HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor_handle_MissionCommand called"));
  api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor (SF (String) &MissionCommandString);

  DeclNewString(_str);
  String__append(SF (String) &_str, string("Received on MissionCommand: "));
  HAMR_SW_RF_Msg_Impl_string_(SF (String) &_str, value);
  api_logInfo__HAMR_SW_ResponseMonitor_Impl_SW_ResponseMonitor_ResponseMonitor(SF (String) &_str);

}
